lib/src/__tests__/DatePickerTestingLib.test.tsx

```diff
@@ -79,6 +91,87 @@ describe('<DatePicker />', () => {
     expect(screen.queryByRole('dialog')).toBeInTheDocument();
   });
 
+  interface FormProps {
+    dateAdapter: new (...args: any) => MuiPickersAdapter<unknown>;
+    locale: any;
+    picker: React.FC<DatePickerProps>;
+    pickerProps: Partial<DatePickerProps>;
+  }
+  const Form: React.FC<FormProps> = ({ dateAdapter, locale, picker: Picker, pickerProps }) => {
+    const [date, setDate] = React.useState<unknown>(new Date('01/01/2020'));
+    return (
+      <LocalizationProvider dateAdapter={dateAdapter} locale={locale}>
+        <Picker
+          onChange={setDate}
+          renderInput={props => <TextField {...props} />}
+          value={date}
+          {...pickerProps}
+        />
+      </LocalizationProvider>
+    );
+  };
+
+  const dateAdapters = {
+    'date-fns': DateFnsUtils,
+    dayjs: DayjsUtils,
+    luxon: LuxonUtils,
+    moment: MomentUtils,
+  };
+
+  const locales = {
+    en: {
+      valid: 'January 2020',
+      invalid: 'Januar 2020',
+    },
+    de: {
+      valid: 'Januar 2020',
+      invalid: 'Janua 2020',
+    },
+  };
+
+  const dateFnsLocales = {
+    en: enLocale,
+    de: deLocale,
+  };
```

Comment:
@oliviertassinari Is this approach in alignment with how you'd like to see the adapters tested in one ci?

Reply:
I guess we need some way to configure the locale for each date library

Reply:
I think only date-fns requires a unique locale.  The others accept a string locale.

Reply:
Checking back here regarding moving this along.  What update would you like to see updated here?  Should we keep with the object of adapters and loop or update the ci?

Reply:
I think that the practical solution is to write the tests once but have the style engine configurable with an env variable, as Dmitriy designed it in the first place, the only improvement opportunity I see is to have all the engines runs in the same build so we can with a single glance run them all in the local env or see all the fails in the CI without opening x windows. 

So regarding this pull request, I think that we should simply configure the locale for each date engine, as you have started with datefns.

Reply:
Also, note that we need to run dayjs in the CI, we didn't made this change yet.

Reply:
Sounds good.  I'll add dayjs to the ci.